### PR TITLE
[#24] [Frontend] As a user, I can apply complex filters to uploaded keywords

### PR DESCRIPTION
**Pull request by Nihisil**
Resolves: https://github.com/Nihisil/nimble-certification-elixir/issues/24

## What happened 👀

Modal window to apply complex filters was added.

### Filters

image

### Results

image

### CODE REVIEW

File: lib/google_scraping/dashboard/keywords.ex [hunk 1]
```diff
@@ -128,22 +128,35 @@ defmodule GoogleScraping.Dashboard.Keywords do
...
+      user_id
+      |> KeywordQuery.user_keyword_urls_stat()
+      |> Repo.aggregate(:count)
+
+    {:ok, result}
+  end
+
+  def apply_filters_to_user_keywords(_user_id, _) do
+    {:error, "Invalid filter"}
   end
```
Comment (bterone) [inline lib/google_scraping/dashboard/keywords.ex:160]: Just single line and specifying the `_` value in the second param
```suggestion
  def apply_filters_to_user_keywords(_user_id, _filter_params),
    do: {:error, "Invalid filter"}
```

**Review by bterone — APPROVED** [review]
Just a minor comment, but ✅ from me 😄

File: test/google_scraping_web/controllers/keyword_filters_controller_test.exs [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+
+  describe "GET index/2" do
+    test "when given filter, renders results count", %{conn: conn} do
+      user = insert(:user)
+
+      _ad_url =
+        insert(:keyword_url, user_id: user.id, url: "https://test.com/technology", is_ad: true)
+
+      _ad_url =
+        insert(:keyword_url, user_id: user.id, url: "https://test.com/some-technology", is_ad: true)
```
Comment (junan) [inline test/google_scraping_web/controllers/keyword_filters_controller_test.exs:12]: We are creating many same `_ad_url`. So how about to use `insert_list`

```suggestion
        insert_list(2, :keyword_url, user_id: user.id, url: "https://test.com/some-technology", is_ad: true)
```

File: lib/google_scraping/dashboard/keywords.ex [hunk 3]
```diff
@@ -128,22 +128,35 @@ defmodule GoogleScraping.Dashboard.Keywords do
...
+  def apply_filters_to_user_keywords(user_id, %{"url_stat" => _value}) do
+    result =
+      user_id
+      |> KeywordQuery.user_keyword_urls_stat()
+      |> Repo.aggregate(:count)
+
+    {:ok, result}
+  end
+
+  def apply_filters_to_user_keywords(_user_id, _filter_params) do
```
Comment (junan) [inline lib/google_scraping/dashboard/keywords.ex:158]: We can do it with one liner syntax

**Review by junan — APPROVED** [review]

**Pull request merged**